resource/stash_client.go

```diff
@@ -50,7 +53,12 @@ func (c *StashClient) SetBuildStatus(commit string, status Status) error {
 	req.Header.Add("content-type", "application/json")
 	req.SetBasicAuth(c.username, c.password)
 
-	resp, err := http.DefaultClient.Do(req)
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: c.skipSSLVerification},
+	}
+	client := &http.Client{Transport: tr}
```

Comment:
Might be worth building and setting the client on the `StashClient` struct. What do you think?